service.py

```diff
@@ -31,82 +31,99 @@
 ICON         = lib.common.ICON
 oldversion = False
 
-monitor = xbmc.Monitor()
 
 class Main:
     def __init__(self):
         linux = False
         packages = []
-
-        if monitor.waitForAbort(5):
-            sys.exit(0)
-
+        global dialogbg
+        dialogbg = xbmcgui.DialogProgressBG()
+        dialogbg.create('Version Check')
         if xbmc.getCondVisibility('System.Platform.Linux') and ADDON.getSetting("upgrade_apt") == 'true':
             packages = ['kodi']
-            _versionchecklinux(packages)
-        else:
-            oldversion, version_installed, version_available, version_stable = _versioncheck()
-            if oldversion:
-                _upgrademessage2( version_installed, version_available, version_stable, oldversion, False)
-                
-def _versioncheck():
-    # initial vars
-    from lib.jsoninterface import get_installedversion, get_versionfilelist
-    from lib.versions import compare_version
-    # retrieve versionlists from supplied version file
-    versionlist = get_versionfilelist()
-    # retrieve version installed
-    version_installed = get_installedversion()
-    # compare installed and available
-    oldversion, version_installed, version_available, version_stable = compare_version(version_installed, versionlist)
-    return oldversion, version_installed, version_available, version_stable
+            self._versionchecklinux(packages)
+        elif ADDON.getSetting("versioncheck_enable") == "true":
```

Comment:
Seems a bit late to check this? So linux would always be checked, even if you disable this setting?

Reply:
In the default version, Linux does not update if there is no new version of Kodi.
The idea is proposed an update of Kodi, or Linux or both elements.
Knowing that on Linux, I am well aware that he will also update Kodi